Dockerfile

```diff
@@ -0,0 +1,26 @@
+FROM node:20-slim AS base
+ENV PNPM_HOME="/pnpm"
+ENV PATH="$PNPM_HOME:$PATH"
+RUN corepack enable
+
+FROM base AS build
+## needed for node-gyp to build
+RUN apt-get update && \
+    apt-get install -y python3 make gcc g++
+COPY . /usr/src/app
+WORKDIR /usr/src/app
+RUN --mount=type=cache,id=pnpm,target=/pnpm/store pnpm install --frozen-lockfile
+RUN pnpm run build
+RUN pnpm deploy --filter=commonwealth --prod /prod/commonwealth
+RUN mv /usr/src/app/packages/commonwealth/build /prod/commonwealth/build
+
+FROM base AS commonwealth
+ENV PORT 8080
```

Comment:
Isn't setting `PORT` here the same as using `EXPOSE 8080` below? I'm confused as to why we need to set a port here at all. `server.js` should default to 8080 or use `PORT` environment variable on Heroku.

Reply:
This pulls it from the ENV file or defaults to 8080. We need to know the port at build-time because we need to have a port mapping to expose that port

Reply:
Ah sorry I think you are right here. It should be ARG instead of ENV. ENV just sets the variables, but ARG gets the passed in variable